cleverhans/experimental/certification/utils.py

```diff
@@ -219,11 +221,11 @@ def tf_lanczos_smallest_eigval(vector_prod_fn,
       dtype, size=1, dynamic_size=True, element_shape=(matrix_dim, 1))
 
   # If start vector is all zeros, make it a random normal vector and run for 1000 iter
-  if tf.reduce_sum(b) == 0:
-    b = tf.random_normal(shape=(matrix_dim, 1), dtype=dtype)
-    max_iter = 1000
+  if tf.norm(initial_vector) < collapse_tol:
```

Comment:
Please update comment above. It's no longer 1000 iterations, but number specified by an argument